sonar-plugin-api/src/main/java/org/sonar/api/platform/Server.java

```diff
@@ -47,6 +45,12 @@
 
   public abstract String getContextPath();
 
+  public abstract String getPublicRootUrl();
```

Comment:
Javadoc is missing, including @since 5.4. Can it be null ? What's the result if the public URL property is not defined ?